packages/fluentui/react-northstar/src/components/Datepicker/Datepicker.tsx

```diff
@@ -213,10 +176,15 @@ export const Datepicker: ComponentWithAs<'div', DatepickerProps> &
 
   const overrideDatepickerCalendarProps = (predefinedProps: DatepickerCalendarProps): DatepickerCalendarProps => ({
     ...calendarOptions,
+    ...dateFormatting,
     onDateChange: (e, itemProps) => {
-      setSelectedDate(itemProps.value.originalDate);
-      setOpen(false);
-      _.invoke(predefinedProps, 'onDateChange', e, itemProps);
+      const targetDay = itemProps.value;
+      setSelectedDate(targetDay.originalDate);
```

Comment:
Please switch state management to `useReducer`, it's hard to follow this

Reply:
Is it then fair to also create a composite state object { error, formattedDate, selectedDate }  or you would keep the states separate?

Reply:
> Is it then fair to also create a composite state object { error, formattedDate, selectedDate } or you would keep the states separate?

Let's keep as it is for now.